flow/src/main/java/flow/Flow.java

```diff
@@ -305,6 +327,14 @@ void enqueue(PendingTraversal pendingTraversal) {
       }
     }
 
+    void bootstrap(History history) {
+      this.nextHistory = checkNotNull(history, "history");
+      if (dispatcher == null) {
+        throw new AssertionError("Bad doExecute method allowed dispatcher to be cleared");
+      }
+      dispatcher.dispatch(new Traversal(null, nextHistory, Direction.REPLACE), this);
```

Comment:
I bet otherwise. In fact, this _cures_ a need for defensive code-- checking for a missing view. Even better, it cures the need to check for a missing view iff the origin and destination histories are equal, a relatively costly test. Instead there's one simple check for `origin == null`. And since origin is annotated `@Nullable` now, AS will highlight with a warning if you don't perform that check.